VictoriaAlbert/VictoriaAlbert/VictoriaAlbertTableViewController.swift

```diff
@@ -0,0 +1,110 @@
+//
+//  VictoriaAlbertTableViewController.swift
+//  VictoriaAlbert
+//
+//  Created by Ana Ma on 11/10/16.
+//  Copyright © 2016 C4Q. All rights reserved.
+//
+
+import UIKit
+
+class VictoriaAlbertTableViewController: UITableViewController, UISearchBarDelegate {
+    
+    let urlString = "http://www.vam.ac.uk/api/json/museumobject/"
+    
+    var objectsArray = [Object]()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        makeSearchBar()
+        APIRequestManager.manager.getData(endpoint: urlString) { (data: Data?) in
+            guard let validData = data else {return}
+            guard let validObjects = Object.getCollectionObject(from: validData) else {return}
+            self.objectsArray = validObjects
+            
+            //            dump(validData)
+            //            dump(self.objectsArray)
+            DispatchQueue.main.async {
+                self.tableView.reloadData()
+            }
+        }
+    }
+    
+    // MARK: - Table view data source
+    
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        return 1
+    }
+    
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        print(objectsArray.count)
+        return objectsArray.count
+    }
+    
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        
+        let cell = tableView.dequeueReusableCell(withIdentifier: "objectCellIdentifier", for: indexPath)
+        
+        let object = objectsArray[indexPath.row]
+        let titleText = "\(object.object), \(object.date_text) - \(object.place)"
+        let detailText = "\(object.title)"
+        
+        cell.textLabel?.text = titleText
+        cell.detailTextLabel?.text = detailText
+        
+        if let validImageURLString = object.imageURLString {
+            APIRequestManager.manager.getData(endpoint: validImageURLString) { (data) in
+                guard let validData = data else {return}
+                
+                DispatchQueue.main.async {
+                    cell.imageView?.image = UIImage(data: validData)
+                }
+            }
+        } else {
+            cell.imageView?.image = UIImage(named: "default-placeholder")
+        }
+        return cell
+    }
+    
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
```

Comment:
You don't need to implement both didSelectRowAt and use a segue. Instead, just create the segue and derive the index to the data by casting the sender to a UITableViewCell. With the cell you can find the indexPath of that cell from the tableView which is a property of the view controller. Then you can index into your data array.